### ISSUE TITLE
Playground Block: A11y: Add text documentation for keyboard exit of CodeMirror

### ISSUE THREAD

**Issue by joedolson** [WordPress/playground-tools#316]
The `esc + tab` key combination for exiting the CodeMirror editor in the Playground block is supported, but not made evident to users. Having a visible text explanation would significantly improve the accessibility of the code view for keyboard users. 

Screen reader users have other ways of exiting the block, but keyboard users have limited options. 

Something to the effect of  "Use `esc` plus `tab` to tab forward out of the code editor." placed before entering the editor should be sufficient.

**Comment by joedolson** [comment]
A slight clarification, since I realize that `esc + tab` technically means something different than the actual mechanism here. The process is to press `esc` and then press `tab` - pressing escape turns off the `tab` listener, so that the tab key will operate normally instead of being captured. It's a little bit different from the usual key combination pattern.

**Comment by brandonpayton** [comment]
Thank you, @joedolson!

@jarekmorawski, we need to add some explanatory text so keyboard users know how to exit the code editor, and I do not currently have any great ideas about where to put it. Given the following UI screenshot, is there anything you might suggest?

![Screenshot 2024-07-23 at 12 09 11 PM](https://github.com/user-attachments/assets/977a40dc-d9a4-4e95-b4e3-29af52ffb73f)

**Comment by jarekmorawski** [comment]
Could something like this work?

<img width="373" alt="image" src="https://github.com/user-attachments/assets/271ee843-7bf6-48a2-bf92-885531e64bb6">

**Comment by joedolson** [comment]
It would be preferable if it was available to users before they entered the block. For low vision users, that text may be very difficult to find, since it may not be in their field of vision.  Additionally, if you've tabbed into the code editor while that region is outside of your viewport, it will be very difficult for you to scroll the viewport into a position where it's visible. E.g.

![image](https://github.com/user-attachments/assets/d2dc81ea-7410-41e9-9cd8-71cec95bf75d)

**Comment by brandonpayton** [comment]
@jarekmorawski @joedolson what if there is a dismissible header between the file tabs and the code editor that explains how to exit the editor with the keyboard? If desired, we could use localStorage to remember it was dismissed so it isn't shown to the same user repeatedly (unless they are browsing in private mode where localStorage isn't persisted across sessions).

**Comment by jarekmorawski** [comment]
Got it. Placing it outside the code editor feels counterintuitive because the block will display the whole Playground site and a button letting you close the code editor will be irrelevant when you're viewing the home page or managing plugins.

I guess we'd try putting it in the top right corner. 

<img width="446" alt="image" src="https://github.com/user-attachments/assets/56b26be6-7e98-414b-b882-27249b407235">

Or, if that doesn't work, maybe we'd show an overlay when the block comes into view, similar to what Chrome does when you make it full-screen. It'd disappear after 800ms. If Google managed to make it accessible, I'm sure we can too.

<img width="805" alt="image" src="https://github.com/user-attachments/assets/ee50cc01-4e97-4792-bf07-618bc2af4a0c">

**Comment by joedolson** [comment]
Chrome's message is not accessible, so that's not really something we can reference. Any notification that disappears without guaranteeing user focus or opportunity to read is a problem; it'll certainly run into problems with [WCAG success criteria 2.2: Enough time](https://www.w3.org/WAI/WCAG21/Understanding/enough-time). 

Top right corner works for me; we'll probably need to make sure there's always enough room for that. For the record, my reference implementation is https://learn.wordpress.org/test/wordpress-playground-block-plugin-test-page/, and as I'm seeing it (screenshot above), it would be pretty hard to fit that into the space.

**Comment by brandonpayton** [comment]
> Top right corner works for me; we'll probably need to make sure there's always enough room for that. For the record, my reference implementation is https://learn.wordpress.org/test/wordpress-playground-block-plugin-test-page/, and as I'm seeing it (screenshot above), it would be pretty hard to fit that into the space.

The fit is my concern as well and is why [I've started wondering](https://github.com/WordPress/playground-tools/issues/316#issuecomment-2250337833) whether we should just make a focusable, dismissible notice that sits between the file tab bar and code editor "Line 1".

**Comment by brandonpayton** [comment]
> The fit is my concern as well and is why https://github.com/WordPress/playground-tools/issues/316#issuecomment-2250337833 whether we should just make a focusable, dismissible notice that sits between the file tab bar and code editor "Line 1".

@jarekmorawski @joedolson do you have any thoughts on this possibility?

**Comment by joedolson** [comment]
I would consider a dismissible notice before entering the code editor to be great. People who don't need it can dismiss it; people who need a reminder can leave it in place; and it has a focusable point so keyboard/low vision navigators have a much better likelihood of finding it.

**Comment by adamziel** [comment]
@brandonpayton what's the status here? Should it be marked as complete? Or punted?

**Comment by brandonpayton** [comment]
@adamziel many of these a11y issues have been addressed but left open until someone with more a11y experience can review them. But I think we can close these and just reopen in response to feedback if needed.

Let's mark this one as complete.

**Comment by joedolson** [comment]
@brandonpayton @adamziel Everything that I checked off in https://github.com/WordPress/Learn/issues/2352#issuecomment-2332416410 can be closed as completed; I didn't make individual comments in each issue.

**Comment by brandonpayton** [comment]
> @brandonpayton @adamziel Everything that I checked off in https://github.com/WordPress/Learn/issues/2352#issuecomment-2332416410 can be closed as completed; I didn't make individual comments in each issue.

Thanks, @joedolson!
